crates/agglayer-certificate-orchestrator/src/network_task.rs

```diff
@@ -255,459 +287,90 @@ where
-                    .await
-                    .map_err(|error| {
-                        error!(hash, "Error while witnessing the execution: {}", error);
-                        error
-                    })?;
-
-                self.pending_state = Some(new_state);
-
-                if let Err(error) = self
-                    .settlement_client
-                    .recover_settlement(eth_tx_hash, certificate_id, self.network_id, header.height)
-                    .await
-                {
-                    error!(hash, "Error while recovering the transaction: {}", error);
-                } else {
-                    info!(
-                        hash,
-                        "Recovered the transaction for certificate {}", certificate_id
-                    );
-
-                    self.handle_certificate_settlement(certificate)?;
-                    *next_expected_height += 1;
-                }
-            } else {
-                // Transaction not found, back to pending
-
-                warn!(
-                    hash,
-                    "Transaction {} not found for certificate {}", eth_tx_hash, hash
-                );
-
-                self.state_store.update_certificate_header_status(
-                    &certificate_id,
-                    &CertificateStatus::Pending,
-                )?;
-                self.pending_store.remove_generated_proof(&certificate_id)?;
-            }
-        } else {
-            // Candidate but no tx_hash, cert should be in error
-            warn!(
-                "Certificate {} is in candidate but has no settlement tx hash",
-                hash
-            );
-
-            self.state_store.update_certificate_header_status(
-                &certificate_id,
-                &CertificateStatus::InError {
-                    error: CertificateStatusError::SettlementError(
-                        "Inconsistent transaction state".to_string(),
-                    ),
-                },
-            )?;
-        }
-
-        Ok(())
-    }
-
-    async fn handle_pending(
-        &mut self,
-        header: CertificateHeader,
-        next_expected_height: &mut u64,
-    ) -> Result<(), Error> {
-        let certificate_id = header.certificate_id;
-        let height = header.height;
-
-        info!(
-            hash = certificate_id.to_string(),
-            "Certifying the certificate {certificate_id} for network {} at height {}",
-            self.network_id,
-            height
-        );
-
-        match self
-            .certifier_client
-            .certify(self.local_state.clone(), self.network_id, height)
-            .await
-        {
-            Ok(CertifierOutput {
-                height,
+        let bridge_exit_hashes = certificate
+            .bridge_exits
+            .iter()
+            .map(|exit| exit.hash())
+            .collect::<Vec<Digest>>();
+        tokio::spawn(
+            CertificateTask::new(
```

Comment:
The handle of the certificate task is immediately dropped. Maybe it should be polled in the loop below so we can detect and log any potential panics. Also maybe await it at the end when the loop exits for the same reason + to have the task lifetimes properly nested which should ease the reasoning about them somewhat.